pkg/agent/agent.go

```diff
@@ -457,15 +461,64 @@ func (c *agentController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 	return ctrl.Result{}, nil
 }
 
-func isVersionHistoryStateFound(history []configv1.UpdateHistory, state configv1.UpdateState) bool {
-	for _, h := range history {
-		if h.State == state {
+func (c *agentController) isHostedControlPlaneAvailable(status hyperv1alpha1.HostedClusterStatus) bool {
+	for _, condition := range status.Conditions {
+		if condition.Reason == hyperv1alpha1.HostedClusterAsExpectedReason && condition.Status == metav1.ConditionTrue && condition.Type == string(hyperv1alpha1.HostedClusterAvailable) {
```

Comment:
This will need to be updated when bumping API hyperv1alpha1.HostedClusterAsExpectedReason -> hyperv1alpha1.AsExpectedReason, you might want to check a fallback check now to ease the transition e.g. `condition.Reason == hyperv1alpha1.HostedClusterAsExpectedReason || "AsExpected"`
Or even be looser for now and only check `condition.Status == metav1.ConditionTrue` until we go v1beta1.

See https://github.com/openshift/hypershift/pull/1887/files#diff-2ea6fd1498b26ef731aca78647490c0151e9d327e3b1801f5aa7611bc759251eR88

Reply:
@enxebre OK So I see that more conditions are added. Instead of hard coding the reason, I opened https://issues.redhat.com/browse/ACM-2167 to handle this when your PR is merged.